src/compute-client/src/plan/interpret/api.rs

```diff
@@ -125,7 +125,14 @@ pub trait Interpreter<T = mz_repr::Timestamp> {
 }
 
 /// An [Interpreter] context.
-pub type Context<Domain> = BTreeMap<LocalId, ContextEntry<Domain>>;
+#[derive(Debug, Default)]
+pub struct InterpreterContext<Domain> {
+    /// The bindings currently in the context.
+    pub bindings: BTreeMap<LocalId, ContextEntry<Domain>>,
+    /// Is the context recursive or not.
```

Comment:
nit: maybe spell this out a bit more clearly:

> Is the context _recursive_ (is one of our ancestors a `LetRec` binding) or not.

Reply:
Yep, done!